### ISSUE TITLE
IsADirectoryError when running r/all on Semgrep itself

### ISSUE THREAD

**Issue by ievans** [semgrep/semgrep#2734]
I see a `IsADirectoryError` when running r/all on Semgrep itself:

```
ine@imbp4 ~/D/r/semgrep (develop) [130]> time semgrep --config=r/all
using config from https://semgrep.dev/r/all. Visit https://semgrep.dev/registry to see all public rules.
downloading config...
running 1082 rules...
 33%|█████████████████████████████████▌                                                                    |356/1082
Traceback (most recent call last):
  File "/usr/local/bin/semgrep", line 33, in <module>
    sys.exit(load_entry_point('semgrep==0.41.1', 'console_scripts', 'semgrep')())
  File "/usr/local/Cellar/semgrep/0.41.1/libexec/lib/python3.9/site-packages/semgrep/__main__.py", line 17, in main
    cli()
  File "/usr/local/Cellar/semgrep/0.41.1/libexec/lib/python3.9/site-packages/semgrep/cli.py", line 441, in cli
    semgrep.semgrep_main.main(
  File "/usr/local/Cellar/semgrep/0.41.1/libexec/lib/python3.9/site-packages/semgrep/semgrep_main.py", line 261, in main
    ) = CoreRunner(
  File "/usr/local/Cellar/semgrep/0.41.1/libexec/lib/python3.9/site-packages/semgrep/core_runner.py", line 551, in invoke_semgrep
    findings_by_rule, debug_steps_by_rule, errors, all_targets = self._run_rules(
  File "/usr/local/Cellar/semgrep/0.41.1/libexec/lib/python3.9/site-packages/semgrep/core_runner.py", line 511, in _run_rules
    rule_matches, debugging_steps, errors, rule_targets = self._run_rule(
  File "/usr/local/Cellar/semgrep/0.41.1/libexec/lib/python3.9/site-packages/semgrep/core_runner.py", line 375, in _run_rule
    self.handle_regex_patterns(outputs, patterns_regex, targets)
  File "/usr/local/Cellar/semgrep/0.41.1/libexec/lib/python3.9/site-packages/semgrep/core_runner.py", line 470, in handle_regex_patterns
    matches = tpool.map(re_fn, targets)
  File "/usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 771, in get
    raise self._value
  File "/usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/usr/local/Cellar/semgrep/0.41.1/libexec/lib/python3.9/site-packages/semgrep/core_runner.py", line 75, in get_re_matches
    contents = path.read_text()
  File "/usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/pathlib.py", line 1255, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "/usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/pathlib.py", line 1241, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
IsADirectoryError: [Errno 21] Is a directory: 'semgrep-core/pfff'

________________________________________________________
Executed in   59.60 mins   fish           external
   usr time   62.36 mins  133.00 micros   62.36 mins
   sys time    2.51 mins  792.00 micros    2.51 mins
```

**Comment by ievans** [comment]
tagging @nbrahms and @underyx who have been talking about potentially recommending r/all

**Comment by mschwager** [comment]
@brendongo could this be related to https://github.com/returntocorp/semgrep-action/issues/227? In `v0.41.1` it looks like `semgrep-core/pfff` is an empty directory, which makes me wonder if something is symlinking to it, or if the directory has other special properties. I.e. why do we get this error on `semgrep-core/pfff`, but no other directories in the tree?

**Comment by ievans** [comment]
confirmed fixed as of 0.56.0
